owncloud/owncloud.py

```diff
@@ -878,6 +919,69 @@ def remove_user_from_group(self, user_name, group_name):
 
         raise ResponseError(res)
 
+    def add_user_to_subadminGroup(self, user_name, group_name):
+        """Adds a user to a subadmin group.
+
+        :param user_name:  name of user to be added to subadmin group
+        :param group_name:  name of subadmin group
+        :returns: True if user added
+        :raises: ResponseError in case an HTTP error status was returned
+
+        """
+
+        res = self.__make_ocs_request(
+            'POST',
+            self.OCS_SERVICE_CLOUD,
+            'users/' + user_name + '/subadmins',
```

Comment:
Not sure how to do it... 
I checked create_group which cannot process group names different than ascii... 
As it's possible to create UTF8 group names in the web interface, I tried to encode string so that I could create an UTF8 group name (Hélicoptère for instance)... sadly it feel like I'm missing something here... 

```

**Request**
POST /ocs/v1.php/cloud/groups HTTP/1.1
Host: depot.dev.local
Content-Length: 29
Accept-Encoding: gzip, deflate
OCS-APIREQUEST: true
Accept: */*
User-Agent: python-requests/2.4.3 CPython/2.7.9 Linux/3.16.0-4-amd64
Connection: keep-alive
Cookie: ockkpar7kpqs=d9b0uiuhk5euckfrchb1pqd0m5
Content-Type: application/x-www-form-urlencoded

groupid=H%C3%A9licopt%C3%A8re


**Response**
HTTP/1.1 200 OK
Date: Mon, 03 Aug 2015 20:51:06 GMT
Server: Apache
Expires: Thu, 19 Nov 1981 08:52:00 GMT
Cache-Control: no-store, no-cache, must-revalidate, post-check=0, pre-check=0
Pragma: no-cache
Content-Security-Policy: default-src 'self'; script-src 'self' 'unsafe-eval'; style-src 'self' 'unsafe-inline'; frame-src *; img-src *; font-src 'self' data:; media-src *; connect-src *
X-XSS-Protection: 1; mode=block
X-Content-Type-Options: nosniff
X-Frame-Options: Sameorigin
X-Robots-Tag: none
Content-Length: 159
Connection: close
Content-Type: text/xml; charset=UTF-8

<?xml version="1.0"?>
<ocs>
 <meta>
  <status>failure</status>
  <statuscode>101</statuscode>
  <message>Invalid group name</message>
 </meta>
 <data/>
</ocs>

``` 

I took some time to see what was going on when I was using the GUI, it's not calling the same location, but encoding might have been similar. 

```
**Request**
POST /index.php/settings/users/groups HTTP/1.1
Host: depot.dev.local
Connection: keep-alive
Content-Length: 24
requesttoken: 8Yql8g5uWHst9Nb6ca5lxaRrwJuYqo
Origin: http://depot.dev.local
CSP: active
User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.130 Safari/537.36
OCS-APIREQUEST: true
Content-Type: application/x-www-form-urlencoded; charset=UTF-8
Accept: */*
X-Requested-With: XMLHttpRequest
Referer: http://depot.dev.local/index.php/settings/users
Accept-Encoding: gzip, deflate
Accept-Language: en-US,en;q=0.8,fr;q=0.6
Cookie: ockkpar7kpqs=4dqum62vp0fkm8v89a1g0v8i06

id=H%C3%A9licopt%C3%A8re

**Response**
HTTP/1.1 201 Created
Date: Mon, 03 Aug 2015 21:04:05 GMT
Server: Apache
Expires: Thu, 19 Nov 1981 08:52:00 GMT
Cache-Control: no-cache, must-revalidate
Pragma: no-cache
Content-Security-Policy: default-src 'none';script-src 'self' 'unsafe-eval';style-src 'self' 'unsafe-inline';img-src 'self';font-src 'self';connect-src 'self';media-src 'self'
X-XSS-Protection: 1; mode=block
X-Content-Type-Options: nosniff
X-Frame-Options: Sameorigin
X-Robots-Tag: none
Content-Length: 37
Connection: close
Content-Type: application/json; charset=utf-8

{"groupname":"H\u00e9licopt\u00e8re"}

```

I tried to override header Content-Type, so that it includes charset=UTF-8; but sadly no quick win. 
I guess I'll have to debug the API too... unless you can give me further instructions. 

cheers